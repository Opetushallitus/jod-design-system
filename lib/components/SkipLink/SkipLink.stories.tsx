import type { Meta, StoryObj } from '@storybook/react';

import { SkipLink } from './SkipLink';

const meta = {
  title: 'SkipLink',
  component: SkipLink,
  tags: ['autodocs'],
} satisfies Meta<typeof SkipLink>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  parameters: {
    docs: {
      description: {
        story: 'Skip link is only visible when getting focus.',
      },
    },
  },
  args: {
    label: 'Skip to main content',
    hash: '#',
  },
};
