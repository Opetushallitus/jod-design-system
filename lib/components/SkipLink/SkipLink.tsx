export interface SkipLinkProps {
  label: string;
  hash: string;
}

export const SkipLink = ({ label, hash }: SkipLinkProps) => {
  return (
    <a
      href={hash}
      className="ds-absolute -ds-left-full ds-z-50 ds-m-5 ds-flex ds-rounded-[4px] ds-border-2 ds-border-secondary-gray ds-bg-white ds-p-5 ds-text-body-md ds-text-button-sm ds-text-link hover:ds-text-accent hover:ds-underline focus:ds-left-auto focus:ds-underline focus-visible:ds-outline focus-visible:ds-outline-[3px] focus-visible:ds-outline-offset-[1.5px]"
    >
      {label}
    </a>
  );
};
