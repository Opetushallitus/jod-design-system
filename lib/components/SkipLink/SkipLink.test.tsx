import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { SkipLink } from './SkipLink';

describe('SkipLink', () => {
  it('renders the label correctly', () => {
    const label = 'Skip to main content';
    const { getByText } = render(<SkipLink label={label} hash={'#'} dataTestId="skip" />);
    expect(document.querySelector('[data-testid="skip"]')).toBeInTheDocument();
    expect(getByText(label)).toBeInTheDocument();
  });

  it('renders the component correctly', () => {
    const label = 'Skip to main content';
    const { container } = render(<SkipLink label={label} hash={'#'} dataTestId="skip2" />);
    expect(document.querySelector('[data-testid="skip2"]')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
