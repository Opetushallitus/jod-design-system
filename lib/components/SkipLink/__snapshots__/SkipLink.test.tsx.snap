// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SkipLink > renders the component correctly 1`] = `
<a
  class="absolute -left-full z-50 flex rounded-[4px] border-2 border-secondary-gray bg-white p-5 text-body-md text-button-sm text-link hover:text-accent hover:underline focus:left-auto focus:underline focus-visible:outline focus-visible:outline-[3px] focus-visible:outline-offset-[1.5px]"
  href="#"
>
  Skip to main content
</a>
`;
